php-frontend/src/main/java/org/sonar/php/metrics/CognitiveComplexityVisitor.java

```diff
@@ -0,0 +1,319 @@
+
+    visitWithNesting(tree.statement());
+    visit(tree.condition());
+  }
+
+  @Override
+  public void visitForStatement(ForStatementTree tree) {
+    complexity.addComplexityWithNesting(tree.forToken());
+
+    tree.init().forEach(this::visit);
+    tree.condition().forEach(this::visit);
+    tree.update().forEach(this::visit);
+    visitWithNesting(tree.statements());
+  }
+
+  @Override
+  public void visitForEachStatement(ForEachStatementTree tree) {
+    complexity.addComplexityWithNesting(tree.foreachToken());
+
+    visit(tree.expression());
+    visit(tree.key());
+    visit(tree.value());
+    visitWithNesting(tree.statements());
+  }
+
+  @Override
+  public void visitCatchBlock(CatchBlockTree tree) {
+    complexity.addComplexityWithNesting(tree.catchToken());
+
+    complexity.incNesting();
+    super.visitCatchBlock(tree);
+    complexity.decNesting();
+  }
+
+  @Override
+  public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
+    complexity.incNesting();
+    super.visitFunctionDeclaration(tree);
+    complexity.decNesting();
+  }
+
+  @Override
+  public void visitFunctionExpression(FunctionExpressionTree tree) {
+    complexity.incNesting();
+    super.visitFunctionExpression(tree);
+    complexity.decNesting();
+  }
+
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree tree) {
+    complexity.incNesting();
+    super.visitMethodDeclaration(tree);
+    complexity.decNesting();
+  }
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpressionTree tree) {
+    complexity.addComplexityWithNesting(tree.queryToken());
+    visit(tree.condition());
+    visitWithNesting(tree.trueExpression());
+    visitWithNesting(tree.falseExpression());
+  }
+
+  @Override
+  public void visitBreakStatement(BreakStatementTree tree) {
+    if (tree.argument() != null) {
+      complexity.addComplexityWithoutNesting(tree.breakToken());
+    }
+    super.visitBreakStatement(tree);
+  }
+
+  @Override
+  public void visitContinueStatement(ContinueStatementTree tree) {
+    if (tree.argument() != null) {
+      complexity.addComplexityWithoutNesting(tree.continueToken());
+    }
+    super.visitContinueStatement(tree);
+  }
+
+  @Override
+  public void visitGotoStatement(GotoStatementTree tree) {
+    complexity.addComplexityWithoutNesting(tree.gotoToken());
+    super.visitGotoStatement(tree);
+  }
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    if (tree.is(CONDITIONAL_AND, CONDITIONAL_OR)) {
+
+      ExpressionTree leftChild = removeParenthesis(tree.leftOperand());
+      ExpressionTree rightChild = removeParenthesis(tree.rightOperand());
+
+      boolean leftChildOfSameKind = leftChild.is(tree.getKind());
+      boolean rightChildOfSameKind = rightChild.is(tree.getKind());
+
+      // For expressions with same-kind operators like "a && (b && c)" we want to have secondary location on leftmost operator
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)